torch/csrc/cuda/Module.cpp

```diff
@@ -883,10 +883,16 @@ static void registerCudaDeviceProperties(PyObject* module) {
       .def_readonly(
           "max_threads_per_multi_processor",
           &cudaDeviceProp::maxThreadsPerMultiProcessor)
+#if USE_ROCM
```

Comment:
It is best to keep our public API the same for all compilation flags.
Can you just make this an empty string here?

Reply:
@albanD ,
Can you please clarify your comment.
Do you mean to remove the `#if USE_ROCM` condition and keep it same for CUDA and ROCm?

Reply:
@albanD How about this proposed solution?  gcnArchName is unconditionally part of the exposed API, but for CUDA builds it defaults to the "name" field.  I couldn't find a solution that made this an empty string since pybind11 is expecting an address of a struct field.

Reply:
Ho you would have to define a global string and get the reference here otherwise yeah.
name sounds fine for now otherwise.